src/abstract_gp.jl

```diff
@@ -200,6 +200,8 @@ function logpdf(f::FiniteGP, Y::AbstractMatrix{<:Real})
     return -((size(Y, 1) * T(log(2π)) + logdet(C)) .+ diag_Xt_invA_X(C, Y .- μ)) ./ 2
 end
 
+Distributions.loglikelihood(f::FiniteGP, y::AbstractVecOrMat{<:Real}) = logpdf(f, y)
+
```

Comment:
This definition seems strange/incorrect since `loglikelihood` is supposed to return always a scalar:
```suggestion
Distributions.loglikelihood(f::FiniteGP, y::AbstractVector{<:Real}) = logpdf(f, y)
Distributions.loglikelihood(f::FiniteGP, y::AbstractMatrix{<:Real}) = sum(logpdf(f, y))
```
However, I'm a bit surprised that you have to define `loglikelihood` at all - the defaults in Distributions should cover it automatically for you (they are basically what I suggested here).

Reply:
Hmmm good point. I'll do some digging. Probably I messed something up.